File: javascript/matrix/matrix.js
export class Matrix {
  constructor(matrixStr) {
    this._matrix = matrixStr
      .split('\n')
      .map(row => row
        .split(' ')
        .map(n => Number.parseInt(n)));
  }

  get rows() {
    return this._matrix;
  }

  get columns() {
    return this._matrix[0]
      .map((_, i) => this._matrix.map(row => row[i]));
  }
}

File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {
    private val matrix: List<List<Int>> = matrixAsString
        .split("\n")
        .map { row ->
            row.trim().split("""\s+""".toRegex())
                .map { valStr -> valStr.toInt() }
        }

    fun column(colNr: Int): List<Int> = matrix.map { it[colNr - 1] }

    fun row(rowNr: Int): List<Int> = matrix[rowNr - 1]
}
